### DISCUSSION TITLE
TargetFramework support

### DISCUSSION THREAD

**Discussion by paulushub** [dotnet/docfx#9429] [Q&A] [answered]
Does this really work? I have tried both .`net7.0` and `net7.0-windows` without success. 
The project includes .NET 6~8 targets support.
```xml
      "src": [
        {
          "files": [
            "SVGImage/DotNetProjects.SVGImage.csproj"
          ],
          "src": "../Source/",
          "properties": {
            "TargetFramework": "net7.0-windows"
          }          
        }
      ],
```
GitHub workflow:
```xml
# Your GitHub workflow file under .github/workflows/
name: DocFX

on:
  push:
    branches:
      - master
      
jobs:
  publish-docs:
    runs-on: windows-latest
    steps:
    - name: Checkout
      uses: actions/checkout@v3
    - name: Dotnet Setup
      uses: actions/setup-dotnet@v3
      with:
        dotnet-version: 7.x

    - run: dotnet tool update -g docfx
    - run: docfx Docs/docfx.json

    - name: Upload DocFX packages
      uses: actions/upload-artifact@v3
      with:
        name: docfx_site
        path: ${{ github.workspace }}/Docs/_site
          
    - name: Deploy
      uses: peaceiris/actions-gh-pages@v3
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
        publish_dir: Docs/_site
```

![image](https://github.com/dotnet/docfx/assets/809351/d0d2cdcd-fec4-4e2e-a879-e886d44dc2ed)

**Comment by KalleOlaviNiemitalo** [comment, marked as answer]
IIRC NuGet package restore uses all `TargetFrameworks`, not only `TargetFramework`.  Then your pipeline has `dotnet-version: 7.x` which doesn't support targeting .NET 8.

But I wonder if you could work around that by setting both  `TargetFrameworks` and `TargetFramework` in DocFX.

**Reply by paulushub** [reply]
Setting the pipeline has `dotnet-version: 8.x` resolved it. Thank you.

**Comment by filzrev** [comment]
>  NuGet package restore uses all TargetFrameworks, not only TargetFramework.

As far as I've tested. 
`dotnet restore` problems can be fixed by modifying following lines

https://github.com/dotnet/docfx/blob/be96e636269cf9fa2b3c6b2f69c33a8a69078285/src/Docfx.Dotnet/DotnetApiCatalog.Compile.cs#L132

**Modified Code**
When `TargetFramework` parameter specified. Pass parameter to `dotnet restore` command.

```
if (msbuildProperties.TryGetValue("TargetFramework", out var targetFramework))
{
    await Process.Start("dotnet", $"restore \"{path}\" -p:TargetFramework={targetFramework} --locked-mode").WaitForExitAsync();
}
else
{
    await Process.Start("dotnet", $"restore \"{path}\"").WaitForExitAsync();
}
```

**Note**
`--locked-mode` settings is required to disable overwrite `packages.lock.json` file.
It's required when target project using `<RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>` settings.

Without `--locked-mode` parameter. `packages.lock.json` is overwritten for specified target framework. 
And cause file diff when executing `docfx metadata` command)
